sklearn/manifold/tests/test_t_sne.py

```diff
@@ -164,7 +164,7 @@ def test_binary_search_neighbors():
     desired_perplexity = 25.0
     random_state = check_random_state(0)
     data = random_state.randn(n_samples, 2).astype(np.float32, copy=False)
-    distances = pairwise_distances(data)
+    distances = pairwise_distances(data).astype(np.float32, copy=False)
```

Comment:
This change is introduced for `_binary_search_perplexity` to run (it does not accept float64 array).

Must `pairwise_distances` preserve `dtype`? If so, should we downcast in this case?

Reply:
> Must pairwise_distances preserve dtype

That's not a simple question :(
Currently it does sometimes but not always. I'm opening an issue about that

Reply:
https://github.com/scikit-learn/scikit-learn/issues/24502

Reply:
Based on discussion in https://github.com/scikit-learn/scikit-learn/issues/24502,  cfc145ab51fcfd7203fd22f8ede08529331c1125 makes `PairwiseDistances` preserves dtypes, hence reverting this now-useless change.